### ISSUE TITLE
compaction_static_shares should be live updatable

### ISSUE THREAD

**Issue by fee-mendes** [scylladb/scylladb#10723]
**Installation details**
Scylla version (or git commit hash): `5.0.rc5-0.20220515.9da666e77` with build-id `5992da65161c733b3f85b8c93bbbb5151d4d321c`

**Description**
One of the things that is often seen in the field is the need to set static shares on the compaction controller via `compaction_static_shares` followed by a rolling restart. This parameter may be changed due to a variety of reasons, such as write heavy workloads being affected due to compaction backlog growing, or during migrations when writes are the vast majority of the workload, among others.

This issue is an enhancement effort to avoid the need for rolling restart a Scylla cluster only to tell compactions to self confine itself and avoid hurting our beloved users CQL primary workload. 

cc @vladzcloudius @raphaelsc

**Comment by vladzcloudius** [comment]
I will ask @igorribeiroduarte  to add it as a follow up of "[scylladb-dev] [PATCH v2 0/6] Make permissions cache live updateable and add an API for resetting authorization cache".

Once the above is merged this should be an easy money for him.

**Comment by mykaul** [comment]
Isn't this fixed via:
```
ommit 07460761fbff69f1d900c9f0384ef5a8ea8b3693
Merge: 752be6536 3b19bcf1a
Author: Pavel Emelyanov <xemul@scylladb.com>
Date:   Tue Jul 19 16:55:55 2022 +0300

    Merge "Make compaction_static_shares and memtable_flush_static_shares live updateable" from Igor Ribeiro Barbosa Duarte (3):

```

?

**Comment by avikivity** [comment]
Fixed by 07460761fbff69f1d900c9f0384ef5a8ea8b3693
